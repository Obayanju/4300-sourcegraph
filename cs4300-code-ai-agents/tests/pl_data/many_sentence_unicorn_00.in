# This problem is taken from Artificial Intelligence, A Modern Approach, 3rd Edition, Chapter Problem 7.2: By Russell and Norvig
# If the unicorn is mythical, then it is immortal,
# but if it is not mythical, then it is a mortal mammal.
# If the unicorn is either immortal or a mammal, then it is horned.
# The unicorn is magical if it is horned.
# 
# Is the unicorn mythical?
# Is the unicorn magical?
# Is the unicorn horned?
# Is the unicorn immortal?
# Is the unicorn mammal?
#
# Truth table
#
# | MYTHICAL | MAGICAL | HORNED | IMMORTAL | MAMMAL | MYTHICAL => IMMORTAL | (!MYTHICAL) => ((!IMMORTAL) & MAMMAL) | (IMMORTAL | MAMMAL) => HORNED | HORNED => MAGICAL | KB |
# +----------+---------+--------+----------+--------+----------------------+---------------------------------------+-------------------------------+-------------------+----+
# |    T     |   T     |   T    |    T     |   T    |          T           |                    T                  |      T                        |         T         |  T |
# |    T     |   T     |   T    |    T     |   F    |          T           |                    T                  |      T                        |         T         |  T |
# |    T     |   T     |   T    |    F     |   T    |        F             |                    T                  |      T                        |         T         | F  |
# |    T     |   T     |   T    |    F     |   F    |        F             |                    T                  |      T                        |         T         | F  |
# |    T     |   T     |   F    |    T     |   T    |          T           |                    T                  |    F                          |         T         | F  |
# |    T     |   T     |   F    |    T     |   F    |          T           |                    T                  |    F                          |         T         | F  |
# |    T     |   T     |   F    |    F     |   T    |        F             |                    T                  |    F                          |         T         | F  |
# |    T     |   T     |   F    |    F     |   F    |        F             |                    T                  |      T                        |         T         | F  |
# |    T     |   F     |   T    |    T     |   T    |          T           |                    T                  |      T                        |       F           | F  |
# |    T     |   F     |   T    |    T     |   F    |          T           |                    T                  |      T                        |       F           | F  |
# |    T     |   F     |   T    |    F     |   T    |        F             |                    T                  |      T                        |       F           | F  |
# |    T     |   F     |   T    |    F     |   F    |        F             |                    T                  |      T                        |       F           | F  |
# |    T     |   F     |   F    |    T     |   T    |          T           |                    T                  |    F                          |         T         | F  |
# |    T     |   F     |   F    |    T     |   F    |          T           |                    T                  |    F                          |         T         | F  |
# |    T     |   F     |   F    |    F     |   T    |        F             |                    T                  |    F                          |         T         | F  |
# |    T     |   F     |   F    |    F     |   F    |        F             |                    T                  |      T                        |         T         | F  |
# |    F     |   T     |   T    |    T     |   T    |          T           |                  F                    |      T                        |         T         | F  |
# |    F     |   T     |   T    |    T     |   F    |          T           |                  F                    |      T                        |         T         | F  |
# |    F     |   T     |   T    |    F     |   T    |          T           |                    T                  |      T                        |         T         |  T |
# |    F     |   T     |   T    |    F     |   F    |          T           |                  F                    |      T                        |         T         | F  |
# |    F     |   T     |   F    |    T     |   T    |          T           |                  F                    |    F                          |         T         | F  |
# |    F     |   T     |   F    |    T     |   F    |          T           |                  F                    |    F                          |         T         | F  |
# |    F     |   T     |   F    |    F     |   T    |          T           |                    T                  |    F                          |         T         | F  |
# |    F     |   T     |   F    |    F     |   F    |          T           |                  F                    |      T                        |         T         | F  |
# |    F     |   F     |   T    |    T     |   T    |          T           |                  F                    |      T                        |       F           | F  |
# |    F     |   F     |   T    |    T     |   F    |          T           |                  F                    |      T                        |       F           | F  |
# |    F     |   F     |   T    |    F     |   T    |          T           |                    T                  |      T                        |       F           | F  |
# |    F     |   F     |   T    |    F     |   F    |          T           |                  F                    |      T                        |       F           | F  |
# |    F     |   F     |   F    |    T     |   T    |          T           |                  F                    |    F                          |         T         | F  |
# |    F     |   F     |   F    |    T     |   F    |          T           |                  F                    |    F                          |         T         | F  |
# |    F     |   F     |   F    |    F     |   T    |          T           |                    T                  |    F                          |         T         | F  |
# |    F     |   F     |   F    |    F     |   F    |          T           |                  F                    |      T                        |         T         | F  |
# +----------+---------+--------+----------+--------+----------------------+---------------------------------------+-------------------------------+-------------------+----+
# True models repeated here:
# +----------+---------+--------+----------+--------+----------------------+---------------------------------------+-------------------------------+-------------------+----+
# |    T     |   T     |   T    |    T     |   T    |          T           |                    T                  |      T                        |         T         |  T |
# |    T     |   T     |   T    |    T     |   F    |          T           |                    T                  |      T                        |         T         |  T |
# |    F     |   T     |   T    |    F     |   T    |          T           |                    T                  |      T                        |         T         |  T |
# +----------+---------+--------+----------+--------+----------------------+---------------------------------------+-------------------------------+-------------------+----+
# The unicorn is Magical and Horned
#
#
tell MYTHICAL => IMMORTAL
tell (!MYTHICAL) => ((!IMMORTAL) & MAMMAL)
tell (IMMORTAL | MAMMAL) => HORNED
tell HORNED => MAGICAL
# questions
dpll MYTHICAL
dpll MAGICAL
dpll HORNED
dpll IMMORTAL
dpll MAMMAL
dpll !MYTHICAL
dpll !MAGICAL
dpll !HORNED
dpll !IMMORTAL
dpll !MAMMAL
