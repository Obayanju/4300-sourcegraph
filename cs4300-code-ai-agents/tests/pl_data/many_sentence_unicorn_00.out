NO, MYTHICAL is not entailed.
YES, MAGICAL is entailed.
YES, HORNED is entailed.
NO, IMMORTAL is not entailed.
NO, MAMMAL is not entailed.
NO, !MYTHICAL is not entailed.
NO, !MAGICAL is not entailed.
NO, !HORNED is not entailed.
NO, !IMMORTAL is not entailed.
NO, !MAMMAL is not entailed.
